src/guidescanpy/commands/add_sam_tag.py

```diff
@@ -0,0 +1,65 @@
+import argparse
+import pysam
+
+supported_tags = ["ce"]
+
+
+def get_parser(parser):
+    parser.add_argument("input", type=str, help="Path to the input sam/bam file.")
+
+    parser.add_argument(
+        "--ce",
+        action="store_true",
+        help="Add cutting-efficiency tag to the sam/bam file.",
+    )
+
+    parser.add_argument("--output", "-o", help="Path to the output sam file.")
+
+    parser.add_argument(
+        "--output-format",
+        choices=["bam", "sam"],
+        help="Specify the output file format: bam or sam",
+        default="bam",
+    )
+    return parser
+
+
+def add_tag(tag_name, input_sam_file, output_sam_file, is_bam):
+    if is_bam:
+        writing_mode = "wb"
+    else:
+        writing_mode = "w"
+
+    with pysam.AlignmentFile(input_sam_file, "r") as input_file, pysam.AlignmentFile(
+        output_sam_file, writing_mode, header=input_file.header
+    ) as output_file:
+        for read in input_file:
+            match tag_name:
+                case "ce":
+                    tag_value = get_ce_value(read)
+                case _:
+                    raise ValueError(
```

Comment:
get rid of this exception, bu have `get_tag_value` raise it if it doesn't recognize the tag.